client/src/screens/Profile/index.js

```diff
@@ -0,0 +1,88 @@
+import React, { Fragment, Component } from 'react';
+import NavBar from '../../components/NavBar';
+import Collapsible from '../../components/Collapsible';
+import { Typography, Grid } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import style from './style';
+import { animateScroll as scroll, Events } from 'react-scroll';
+import Content from '../../Content/firstPage';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import { IconButton } from '@material-ui/core';
+
+class Profile extends Component {
+	state = {
+		data: [],
+	};
+	componentDidMount() {
+		Events.scrollEvent.register('begin', function () {
+			console.log('begin', arguments);
+		});
+
+		Events.scrollEvent.register('end', function () {
+			console.log('end', arguments);
+		});
+		return this.setState({ data: Content() });
+	}
+	updateState = () => {
+		return this.setState({ data: Content() });
+	};
+	scrollToTop = () => {
+		scroll.scrollToTop();
+	};
+
+	handleClickOpen = (id) => (event) => {
+		if (id === 1) this.props.history.push('/user/:userId/personal-info');
+		else if (id === 3) this.props.history.push('/user/:userId/accounts');
+		else if (id === 4) this.props.history.push('/user/:userId/projects');
+		else if (id === 5) this.props.history.push('/user/:userId/submitted-form');
+		event.preventDefault();
```

Comment:
This is so nice and symmetrical!

What I'm about to show is probably going to seem cRaZy compared to how nice pristine this currently is but:

```js
// Constants
const ROUTE_ID_MAP = {
  PERSONAL_INFO: 1,
  ACCOUNTS: 3,
  PROJECTS: 4,
  SUBMITTED_FORM: 5,
}

const ROUTE_MAP = {
  PERSONAL_INFO: '/user/:userId/personal-info',
  ACCOUNTS: '/user/:userId/accounts',
  PROJECTS: '/user/:userId/projects',
  SUBMITTED_FORM: '/user/:userId/submitted-form',
}

...

// Helpers
const getRouteName = (id) => {
  const [matchingRouteKey, _id] = Object.entries(ROUTE_ID_MAP).find((key, value) => id === value)
  return matchingRouteKey
}

const getRoutePathByName = (routeName) => ROUTE_MAP[routeKey]

...

// Component logic
handleClickOpen = (id) => (event) => {
  const routeName = getRouteName(id)
  const routePath = getRoutePathByName(routeName)

  this.props.history.push(routePath)
}

```


So I don't actually recommend this alternative in this case. But why am I showing you it? 

In the context of the `handleClick` method, if/else is a lovely way to handle mapping the id's to the routes.

But the trouble is that conditional logic is tightly coupled to this interface. Chances are there will be other parts  of the app that need to map IDs again.

Having to remember which ID maps to what entity introduces risk -- so I wanted to show an example of how to abstract the conditional logic into config. As much as possible "config" (maps etc) is where conditional logic should live.

Reply:
OMG super genius